polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -2421,17 +2422,16 @@ impl<T: Config> Pallet<T> {
 		mut message: Xcm<()>,
 	) -> Result<XcmHash, SendError> {
 		let interior = interior.into();
+		let origin_dest = interior.clone().into();
 		let dest = dest.into();
-		let maybe_fee_payer = if interior != Junctions::Here {
-			message.0.insert(0, DescendOrigin(interior.clone()));
-			Some(interior.into())
-		} else {
-			None
-		};
+		let is_waived = <T::XcmExecutor as FeeManager>::is_waived(Some(&origin_dest), FeeReason::ChargeFees);
+		if !is_waived {
+			message.0.insert(0, DescendOrigin(interior));
+		}
 		log::debug!(target: "xcm::send_xcm", "dest: {:?}, message: {:?}", &dest, &message);
 		let (ticket, price) = validate_send::<T::XcmRouter>(dest, message)?;
-		if let Some(fee_payer) = maybe_fee_payer {
-			Self::charge_fees(fee_payer, price).map_err(|_| SendError::Fees)?;
+		if !is_waived {
+			Self::charge_fees(origin_dest, price).map_err(|_| SendError::Fees)?;
 		}
```

Comment:
This condition is good! 👍 

Just rename the var

```suggestion
		if !is_waived {
			Self::charge_fees(local_origin, price).map_err(|_| SendError::Fees)?;
		}
```